Mathlib/Algebra/Group/Translate.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.BigOperators.Pi
+import Mathlib.Data.Set.Pointwise.SMul
+
+/-!
+# Translation operator
+
+This file defines the translation of a function from a group by an element of that group.
+
+# Notation
+
+`τ a f` is notation for `translate a f`.
+-/
+
+open Function Set
+open scoped Pointwise
+
+variable {ι α β M G H R : Type*} [AddCommGroup G]
+
+/-- Translation of a function in a group by an element of that group.
+`τ a f` is defined as `x ↦ f (x - a)`.  -/
+def translate (a : G) (f : G → α) : G → α := fun x ↦ f (x - a)
+
+@[inherit_doc] notation "τ " => translate
+
+@[simp] lemma translate_apply (a : G) (f : G → α) (x : G) : τ a f x = f (x - a) := rfl
+@[simp] lemma translate_zero (f : G → α) : τ 0 f = f := by ext; simp
+
+lemma translate_translate (a b : G) (f : G → α) : τ a (τ b f) = τ (a + b) f := by
+  ext; simp [sub_sub]
+
+lemma translate_add (a b : G) (f : G → α) : τ (a + b) f = τ a (τ b f) := by ext; simp [sub_sub]
+
+lemma translate_add' (a b : G) (f : G → α) : τ (a + b) f = τ b (τ a f) := by
+  rw [add_comm, translate_add]
+
+lemma translate_comm (a b : G) (f : G → α) : τ a (τ b f) = τ b (τ a f) := by
+  rw [← translate_add, translate_add']
+
+@[simp] lemma comp_translate (a : G) (f : G → α) (g : α → β) : g ∘ τ a f = τ a (g ∘ f) := rfl
```

Comment:
This is asserting that simp-normal-form for τ is as far out as possible - let's add a comment to this effect.